mongodb/activity/queryDocument/metadata.go

```diff
@@ -0,0 +1,93 @@
+package queryDocument
+
+import (
+	"github.com/project-flogo/core/data/coerce"
+	"github.com/project-flogo/core/support/connection"
+	mongocon "github.com/project-flogo/datastore-contrib/mongodb/connector/connection"
+)
+
+// Settings structure
+type Settings struct {
+	//JSONDocument interface{}        `md:"jsonDocument,required"`
+	Connection connection.Manager `md:"mongoConnection,required"`
+	Operation  string             `md:"operation,required"`
+	CollName   string             `md:"collName,required"`
+}
+
+//Input structure
+type Input struct {
+	Input map[string]interface{} `md:"input,required"`
```

Comment:
Question, can't this be just JsonDocument instead of Input ?

Reply:
I guess most of our FE connectors follow the pattern where we have an uber input and output complex structure to box in every other input and output elements. We simply followed that for consistency.

Reply:
I understand that, but this is OSS. It would be unnecessary and confusing to have `input` section inside of an "input".  There are cases where there's no other way, but I'm not sure if this falls under that. @vijaynalawade thoughts ?

Reply:
@skothari-tibco : What are you suggesting to do here instead?

Reply:
Changing the filed name `input` to `jsonDocument` or something along that line.  So instead of having 
``` 
"input" : {
  "input" : {
       "jsonDocument" :{}
  }
}
```
Just :  
```
  "input" : {
       "jsonDocument":{}
  }
```

Reply:
Based on configuration , it seems to me that it will be always :
```
"input" : {
  }
```

Reply:
Didn't quite understand what you mean.

Reply:
@skothari-tibco I get what you are saying with respect to Input. I will need to modify the input and output schema for all activities and will need to change my UI typescript code and runtime go code. QA won't like it at this point of dev cycle as they will have to retest all the activities that were given to them via our build system. It may not be much work for us devs but for QA, they are looking at extending their efforts by at least a few days. With Code freeze in about a week, they will ask us to push out the release, which is not ideal for this issue as this is not a blocker or critical issue. I will make these changes post our FE MongoDB 1.0.0 release and will push my changes into OSS and have it ready to be made available in FE as well for the next release of MongoDB (after 1.0.0 release)

Reply:
@fm-tibco @mellistibco  thoughts ?

Reply:
A few things to note:
- If we make changes to input schema format after the 1.0.0 release, as I have suggested, it will most likely have impact on backward compatibility post 1.0.0
- Though OSS doesn't really care about how downstream uses the Flogo platform, the pattern that we FE connectors use for building out input schema, where we don't declare the input schema elements in the descriptor.json and rather declare the complex structures in a separate JSON and have the ts code read from there, does not really allow us to have the structure that you are suggesting..

Reply:
We should avoid backward incompatible changes as far as possible. Though, I don't see any harm in keeping it input.

Reply:
Actually, I agree with Samip.  I would like to understand this a bit more.  What is the input here?  A json that you are inserting?

Reply:
Also maybe we shouldn't make this a "1.0.0" release.

Reply:
Modified the input structure as per the comment provided so that it is now 
 "input" : {
       "jsonDocument":{}
  }